### ISSUE TITLE
Error talking to Funnel: confusing ExecutorLog and TaskLog when logs has an empty entry

### ISSUE THREAD

**Issue by adamnovak** [ohsu-comp-bio/py-tes#31]
I've started trying to use `py-tes==0.4.1` to add TES client support to UCSC's Toil workflow runner. I'm developing against Funnel, which reports service info of:

```
ServiceInfo(name='Funnel', doc='git commit: 43b289a\ngit branch: master\ngit upstream: git@github.com:ohsu-comp-bio/funnel.git\nbuild date: 2020-12-09T21:52:48Z\nversion: 0.10.1', storage=None)
```

I'm able to launch tasks, but when I go to poll the tasks with `get_task()` I get internal errors from py-tes complaining they can't be deserialized:

```
 {'id': 'c55qjplpsjir0oo1kdj0', 'state': 'QUEUED', 'name': 'toil-bbc72af7-e11a-4831-9392-669ea6c309a1-0', 'executors': [{'image': 'quay.io/ucsc_cgl/toil:5.5.0a1-6b31ed0bffc4a28d0450dd4da90db32f26a11fbc-py3.7', 'command': ['_toil_kubernetes_executor', 'gAWVGwAAAAAAAAB9lIwHY29tbWFuZJSMCnNsZWVwIDEwMDCUcy4=']}], 'logs': [{}], 'creationTime': '2021-09-22T18:10:46.934702072-04:00'} could not be unmarshalled to type: Task
            TypeError: ("'logs' must be a list of <class 'tes.models.TaskLog'> (got [ExecutorLog(start_time=None, end_time=None, stdout=None, stderr=None, exit_code=None)] that is a list of <class 'tes.models.ExecutorLog'>).", Attribute(name='logs', default=None, validator=<optional validator for _ListOfValidator(type=<class 'tes.models.TaskLog'>) or None>, repr=True, eq=True, order=True, hash=None, init=True, metadata=mappingproxy({}), type=None, converter=None, kw_only=False, inherited=False, on_setattr=None), <class 'tes.models.TaskLog'>, [ExecutorLog(start_time=None, end_time=None, stdout=None, stderr=None, exit_code=None)])
```

I think it is seeing an empty object in the `logs` array on the task, deciding it must be an `ExecutorLog`, and then complaining that it isn't a `TaskLog` like it should be and it's giving the list the wrong type.

**Comment by adamnovak** [comment]
The problem is here:

https://github.com/ohsu-comp-bio/py-tes/blob/7606dc8c74fb69ac65e1fb3acf8c16aa4cb42d9e/tes/utils.py#L67-L73

And here:

https://github.com/ohsu-comp-bio/py-tes/blob/7606dc8c74fb69ac65e1fb3acf8c16aa4cb42d9e/tes/utils.py#L49

All objects in fields named `logs` are deserialized by first trying to read them as an `ExecutorLog`, and only then trying to read them as a `TaskLog`, without regard for what type is actually expected. An empty object `{}` deserializes fine as either, since the fields are optional, so it always becomes an `ExecutorLog`, even when it is supposed to be a `TaskLog`.

I think the solution is to consult `o` (the expected type we are trying to deserialize) when building `omap`, so we can look for a `logs` field of the type appropriate to the parent type.

**Comment by kellrott** [comment]
Fixed in #32
